### PR TITLE
cors: refactor loop to if

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: cors: refactor loop to if
Additional Description:
PR #36079 updated the policies of the CORS filter to be a vector of references that must exist.
This followup changes the original for-loop to an if statement that returns the first element, if exists.

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: source/extensions/filters/http/cors/cors_filter.cc [hunk 1]
```diff
@@ -292,18 +292,10 @@ bool CorsFilter::allowPrivateNetworkAccess() {
...
 bool CorsFilter::shadowEnabled() {
-  for (const Router::CorsPolicy& policy : policies_) {
-    return policy.shadowEnabled();
-  }
-  return false;
+  return policies_.empty() ? false : policies_[0].get().shadowEnabled();
 }
 
-bool CorsFilter::enabled() {
-  for (const Router::CorsPolicy& policy : policies_) {
```
Comment (adisuissa) [inline source/extensions/filters/http/cors/cors_filter.cc:302]: Question for the filter maintainers (@wbpcode @daixiang0):
Should the enabled/shadowEnabled return whether the first policy (if exists) is enabled, or the OR of all the policies?

Comment (adisuissa) [reply source/extensions/filters/http/cors/cors_filter.cc:302]: FWIW, when following the code paths, it seems that policies will contain all the Cors-policies in the order determined by [perFilterConfigs()](https://github.com/envoyproxy/envoy/blob/7cf53442e41632514030cd4b662e94ada30bea2d/envoy/http/filter.h#L465-L470).
With these overrides it is not immediately clear when a policy should be enabled or not, but if I had to guess, I'd say that if the inner-most (last) policy is enabled, then the filter should be enabled. If that's the case I'll need to update the PR to return the last element (around a runtime guard).

Comment (wbpcode) [reply source/extensions/filters/http/cors/cors_filter.cc:302]: The first policy is the most specific policy.

```
  policies_ = Http::Utility::getAllPerFilterConfig<Router::CorsPolicy>(decoder_callbacks_);

  // The 'perFilterConfigs' will handle cors policy of virtual host first. So, we need
  // reverse the 'policies_' to make sure the cors policy of route entry to be first item in the
  // 'policies_'.
  if (policies_.size() >= 2) {
    std::reverse(policies_.begin(), policies_.end());
  }
```

**Review by wbpcode — DISMISSED** [review]
LGTM. Thanks.

File: source/extensions/filters/http/cors/cors_filter.cc (hunk 1 above)
Comment (adisuissa) [reply source/extensions/filters/http/cors/cors_filter.cc:302]: Ah, I missed that part. Thanks for the clarification!
I'll add comments to the functions, easing the reading of this code.

**Pull request merged**